src/nvim/lib/khash.h

```diff
@@ -171,17 +173,32 @@ typedef khint_t khiter_t;
 #define kroundup32(x) (--(x), (x)|=(x)>>1, (x)|=(x)>>2, (x)|=(x)>>4, (x)|=(x)>>8, (x)|=(x)>>16, ++(x))
 #endif
 
-#ifndef kcalloc
-#define kcalloc(N,Z) xcalloc(N,Z)
-#endif
-#ifndef kmalloc
-#define kmalloc(Z) xmalloc(Z)
-#endif
-#ifndef krealloc
-#define krealloc(P,Z) xrealloc(P,Z)
-#endif
-#ifndef kfree
-#define kfree(P) xfree(P)
+#ifdef USE_LIBC_ALLOCATOR
+# ifndef kcalloc
+#  define kcalloc(N,Z) calloc(N,Z)
+# endif
+# ifndef kmalloc
+#  define kmalloc(Z) malloc(Z)
+# endif
+# ifndef krealloc
+#  define krealloc(P,Z) realloc(P,Z)
```

Comment:
As a note: 
There is a difference in behavior for `xrealloc(not NULL ,0)` and `realloc(not NULL, 0)`.
`realloc()` acts here as `free()`, `xrealloc()` will always allocate at least one byte.